src/components/IndividualCategory.jsx

```diff
@@ -0,0 +1,113 @@
+import { useState, useEffect } from "react";
+import { useParams, Link, useNavigate } from "react-router-dom";
+import { getCategories, getReviews } from "../utils/API";
+import ThumbsCategory from "./ThumbsCategory";
+
+const IndividualCategory = () => {
+  const { slug } = useParams();
+  const [singleCategory, setSingleCategory] = useState({});
+  const [categoryReviews, SetCategoryReviews] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [notFound, setNotFound] = useState(false);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    getCategories().then((response) => {
+      const filteredCategory = response.filter(
+        (category) => category.slug === slug
+      );
+      if (filteredCategory.length === 0) {
+        setNotFound(true);
+      }
+      setSingleCategory(filteredCategory[0]);
+      setLoading(false);
+    });
+  }, [slug]);
+
+  useEffect(() => {
+    getReviews().then((response) => {
+      const filteredReviews = response.reviews.filter(
+        (review) => review.category === slug
+      );
+      SetCategoryReviews(filteredReviews);
+      setLoading(false);
+    });
+  }, [slug]);
+
+  if (notFound) {
+    return (
+      <div>
+        <h2>Error 404: Category Not Found</h2>
+        <button onClick={() => navigate("/categories")}>
+          Go to categories
+        </button>
+      </div>
```

Comment:
This UI is very similar to the `ErrorHandling` component. Consider re-using that component rather than repeating it. This helps keep the app extensible and you can pass props to the component if you need to customise it slightly.